ramanujantools/matrix.py

```diff
@@ -74,19 +74,30 @@ def subs(self, substitutions: Dict) -> Matrix:
             return self.numerical_subs(substitutions)
         return self.xreplace(substitutions)
 
-    def _is_numeric(self, substitutions: Dict) -> bool:
+    def _can_call_flint_walk(self, trajectory: Dict, start: Dict) -> bool:
+        return (
+            self.free_symbols - set(start.keys()).union(trajectory.keys()) != set()
```

Comment:
What if we remove all free_symbols, but subtract a new symbol inside?
In theory, we should support it.